cvmfs/cache_ram.cc

```diff
@@ -116,21 +174,28 @@ int RamCacheManager::Dup(int fd) {
  */
 int RamCacheManager::Readahead(int fd) {
   ReadLockGuard guard(rwlock_);
-  if (!IsValid(fd))
+  if (!IsValid(fd)) {
+    LogCvmfs(kLogCache, kLogDebug, "bad fd %d on Readahead", fd);
     return -EBADF;
+  }
+  LogCvmfs(kLogCache, kLogDebug, "readahead (no-op) on %d", fd);
+  perf::Inc(counters_.n_readahead);
   return 0;
 }
 
 
 int RamCacheManager::StartTxn(const shash::Any &id, uint64_t size, void *txn) {
+  LogCvmfs(kLogCache, kLogDebug, "new transaction with id %s",
+           id.ToString().c_str());
   Transaction *transaction = new (txn) Transaction();
   transaction->id = id;
   transaction->pos = 0;
   transaction->expected_size = size;
   transaction->size = (size == kSizeUnknown) ? kPageSize : size;
   if (transaction->size) {
-    transaction->buffer = smmap(transaction->size);
+    transaction->buffer = scalloc(1, transaction->size);
```

Comment:
I see the reason further down.  Never mind.  It could be an uninitialized `smalloc` though, no?